File: problems/0941-Valid-Mountain-Array/sol1.cpp
// ==========================================================
// 941. Valid Mountain Array
// Difficulty : Easy
// Language   : C++
// Solution   : #1
// Runtime    : 19 ms (Beats 66%)
// Memory     : 26.4 MB (Beats 21%)
// Link       : https://leetcode.com/problems/valid-mountain-array/
// ==========================================================

class Solution {
public:
    bool validMountainArray(vector<int>& arr) {
        
        int n=arr.size();
        int i=0;
        while(i+1<n && arr[i]<arr[i+1])
        {
            i++;

        }
         if(i==0 || i==n-1)
            {
                return false;
            }
         while(i+1<n && arr[i]>arr[i+1])
         {
                i++;

                
         }



        
          if(i==n-1)
                {
                    return true;
                }

         return false;
    }
    
};

File: problems/0134-Gas-Station/sol1.py
# ==========================================================
# 134. Gas Station
# Difficulty : Medium
# Language   : Python
# Solution   : #1
# Runtime    : 19 ms (Beats 76%)
# Memory     : 26.1 MB (Beats 15%)
# Link       : https://leetcode.com/problems/gas-station/
# ==========================================================

class Solution:                                                          
    def canCompleteCircuit(self, gas: List[int], cost: List[int]) -> int:
        if sum(gas) < sum(cost):
            return -1           
        start = 0
        tank = 0 
        for i in range(len(gas)):         
            tank = tank + gas[i] - cost[i]
            if tank < 0:     
                tank = 0     
                start = i + 1
        return start          